lib/streamlit/runtime/runtime.py

```diff
@@ -632,6 +595,9 @@ def _send_message(self, session_info: SessionInfo, msg: ForwardMsg) -> None:
                 session_info.session, session_info.script_run_count
             )
 
+        assert (
+            session_info.client is not None
+        ), "SessionClient should never be None here!"
```

Comment:
See https://github.com/streamlit/streamlit/pull/5422/files#r980846794 for an explanation of why we're somewhat jankily enforcing this invariant this way for now.